.github/workflows/new-release.yml

```diff
@@ -56,19 +60,29 @@ jobs:
         run: |
           NX_URL_GITHUB_GO_CURRENT_VERSION="https://github.com/${{ github.repository }}/commit/$(git rev-parse HEAD)" \
           yarn run nx run frontend:build-vite:production
-      - name: Deploy to Github Pages
+      - name: Try deploy to Github Pages
         run: |
           git fetch origin gh-pages --depth=1
           git worktree add ../gh-pages gh-pages --no-checkout
           cp -rT dist/apps/frontend ../gh-pages
           cd ../gh-pages
           git add --all .
           git commit -m "${{ steps.get-commit-msg.outputs.msg }}"
-          git push
-      - name: Push
+          if ! git push origin; then
+            git push origin HEAD:actions/deploy-gh-pages --force
+            gh pr create --base gh-pages --head actions/deploy-gh-pages --title '${{ steps.get-commit-msg.outputs.title }}' --body ''
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
+      - name: Try push
         run: |
           git status
-          git push
+          if ! git push origin; then
+            git push origin HEAD:actions/new-release --force
+            gh pr create --base $(git rev-parse --abbrev-ref HEAD) --head actions/new-release --title '${{ steps.get-commit-msg.outputs.title-master }}' --body ''
+          fi
+        env:
+          GH_TOKEN: ${{ github.token }}
       - name: Make release
```

Comment:
Don't we need to move this to a `workflow_run` action that runs when the master-pr completes?

Reply:
Oops, I forgot we had ff merges disabled